### PR TITLE
Location and Campus swap buttons now dynamic

### PR DESCRIPTION
**Pull request by MichelRahme**
# Description

When the directions drawer is closed, the buttons will shift up, and when the drawer is open they will shift back down.

pull1

## Related Issues

#138

### CODE REVIEW

File: lib/providers/map_data.dart [hunk 1]
```diff
@@ -12,6 +13,8 @@ class MapData extends ChangeNotifier {
   PanelController panelController = new PanelController();
   String controllerStarting;
   String controllerDestination;
+  double swapButtonTop = SizeConfig.safeBlockVertical * 66;
+  double locationButtonTop = SizeConfig.safeBlockVertical * 75;
```
Comment (EvanMateo7) [inline lib/providers/map_data.dart]: This is causing the failure of unit tests. I don't think SizeConfig should be referenced here. Instead make SizeConfig as a separate Provider. Having it in MapData creates coupling.

Comment (EvanMateo7) [reply lib/providers/map_data.dart]: Also note SizeConfig is using MediaQuery and so when testing MapData it requires an initialized SizeConfig and therefore a MediaQuery.

Comment (MichelRahme) [reply lib/providers/map_data.dart]: Fixed

File: lib/widgets/map_widget.dart [hunk 2]
```diff
@@ -92,11 +92,13 @@ class _MapWidgetState extends State<MapWidget> {
...
     }
-    SizeConfig().init(context);
     final _completer = Provider.of<MapData>(context).getCompleter;
     final _buildings = Provider.of<BuildingsData>(context);
     final pos = Provider.of<UserLocation>(context);
-
+    if (sizeConfig) {
+      SizeConfig().init(context);
+      sizeConfig = false;
+    }
```
Comment (EvanMateo7) [inline lib/widgets/map_widget.dart]: Probably could just initialize SizeConfig in HomePage() instead.

Comment (MichelRahme) [reply lib/widgets/map_widget.dart]: You're absolutely right. Haha

**Pull request merged**